src/OmniSharp.Roslyn.CSharp/CSharpLanguage.cs

```diff
@@ -0,0 +1,15 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using OmniSharp.Mef;
+
+namespace OmniSharp.Roslyn.CSharp.Services
+{
+    public class LanguagePredicate
+    {
+        private static readonly string[] ValidCSharpExtensions = { "cs", "csx", "cake" };
+        [OmniSharpLanguage(LanguageNames.CSharp)]
+        public Func<string, Task<bool>> IsApplicableTo { get; } = filePath => Task.FromResult(ValidCSharpExtensions.Any(extension => filePath.EndsWith(extension)));
```

Comment:
This was impossible to do as a method, and it's specifically something that isn't supported by the PCL version of MEF.  Delegate Properties on the other hand work just fine.